core/trino-main/src/main/java/io/trino/operator/FlatHashStrategyCompiler.java

```diff
@@ -463,11 +453,18 @@ private static MethodDefinition generateHashBlockVectorized(ClassDefinition defi
             rleHandling.append(invokeStatic(Arrays.class, "fill", void.class, hashes, constantInt(0), length, hash));
         }
         else {
-            rleHandling.append(new ForLoop("for (int index = 0; index < length; index++) { hashes[index] = CombineHashFunction.getHash(hashes[index], hash); }")
-                    .initialize(index.set(constantInt(0)))
-                    .condition(lessThan(index, length))
-                    .update(index.increment())
-                    .body(setHashExpression));
+            // CombineHashFunction.combineAllHashesWithConstant(hashes, 0, length, hash)
+            rleHandling.append(invokeStatic(CombineHashFunction.class, "combineAllHashesWithConstant", void.class, hashes, constantInt(0), length, hash));
+        }
+
+        BytecodeExpression setHashExpression;
+        if (field.index() == 0) {
+            // hashes[index] = hash;
+            setHashExpression = hashes.setElement(index, hash);
+        }
+        else {
+            // hashes[index] = CombineHashFunction.getHash(hashes[index], hash);
+            setHashExpression = hashes.setElement(index, invokeStatic(CombineHashFunction.class, "getHash", long.class, hashes.getElement(index), hash));
```

Comment:
Why did this move?

Reply:
Previously, the same hash assignment expression was used in the RLE loop as well as the normal block loop, so it was declared above both of them (RLE loop was defined first). Now it's only used in the normal block loop, so I moved it closer to it's usage site.